library/L2_HAL/sensors/environment/light/temt6000x01.hpp

```diff
@@ -13,33 +13,32 @@ class Temt6000x01 final : public LightSensor
 {
  public:
   /// @param adc The ADC peripheral used to read the sensor data.
-  /// @param adc_reference_voltage The reference voltage of the ADC. This value
-  ///                              will vary based on the chosen MCU.
+  /// @param supply_voltage The voltage supplied to this device.
   /// @param pull_down_resistance The resistance of the pull down resistor used
   ///                             in the circuit.
   explicit constexpr Temt6000x01(sjsu::Adc & adc,
-                                 units::voltage::volt_t adc_reference_voltage,
+                                 units::voltage::volt_t supply_voltage,
                                  units::impedance::ohm_t pull_down_resistance)
       : adc_(adc),
-        kAdcReferenceVoltage(adc_reference_voltage),
-        kPullDownResistance(pull_down_resistance),
-        adc_resolution_(0)
+        kSupplyVoltage(supply_voltage),
+        kPullDownResistance(pull_down_resistance)
   {
-    adc_resolution_ = static_cast<float>((1 << adc_.GetActiveBits()) - 1);
   }
+
   /// Initializes the ADC driver.
   ///
-  /// @return The initialization status.
+  /// @returns The initialization status.
   Status Initialize() const override
   {
     return adc_.Initialize();
   }
-  /// @return The illuminance in units of lux ranging from 1 - 1'000.
+
+  /// @returns The illuminance in units of lux ranging from 1 - 1'000.
   units::illuminance::lux_t GetIlluminance() const override
   {
-    // voltage = adc_output * (adc_reference_voltage / adc_resolution)
+    // voltage = reference_voltage * (adc_output / adc_resolution)
     const units::voltage::microvolt_t kVoltage(
-        adc_.Read() * kAdcReferenceVoltage / adc_resolution_);
+        kSupplyVoltage * adc_.Read() / adc_.GetMaximumValue());
```

Comment:
After some thought about this, I realized that I was wrong. If all you need to do for this driver is derive the current, this can be done without the need of a supply voltage variable. We only need the absolute voltage across the pull down resistor to figure out the amount of current passing through the light sensor. So you can revert this back to simply using `adc.Voltage()`.

Reply:
Revert the change back and you can submit this.